File: 392.is-subsequence.py
#
# @lc app=leetcode id=392 lang=python
#
# [392] Is Subsequence
#

# @lc code=start
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        index = -1
        for c in s:
            try:
                index = t.index(c, index+1)
            except ValueError:
                return False
        return True
# @lc code=end

File: 392.is-subsequence.cpp
/*
 * @lc app=leetcode id=392 lang=cpp
 *
 * [392] Is Subsequence
 */
#include <string>
using namespace std;

// @lc code=start
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int index = -1;
        for (int i = 0; i < s.size(); i++) {
            index = t.find(s[i], index+1);
            if (index == string::npos)
                return false;
        }
        return true;
    }
};
// @lc code=end

